### PR TITLE
Prevent lineWidth object from filling the cell in black [#897]

### LINKED ISSUE #897: fillColor turn's black when specifing an object to lineWidth
This happens when you assign the object with some or all properties (top, right, bottom, left).

The workaround is to assign fillColor a value that is not false or null (another issue), in my case I had to assign white.

### PR DESCRIPTION
**Pull request by MikeDabrowski**
Completely reworked the way the border drawing works. It now takes into consideration the neighbouring borders to achieve 'sharp' edge. Borders are still centered on cell edge.

If `lineWidth: number` is provided, borders will be drawn using rectangle with stroke.

If `lineWidth: Partial<LineWidth>` is provided, borders will be drawn individually.

The tests passed, and from what I could see, all examples are working. I've also added example page with borders showcase - it may not be beautiful but showcases what can be done with the borders now.

Closes #897

### CODE REVIEW

File: src/tableDrawer.ts [hunk 1]
```diff
@@ -392,9 +392,12 @@ function drawCellBorders(doc: DocHandler, cell: Cell, cursor: Pos) {
       doc.rect(cell.x, cursor.y, cell.width, cell.height, fillStyle)
     }
   } else if (typeof cellStyles.lineWidth === 'object') {
-    doc.rect(cell.x, cursor.y, cell.width, cell.height, 'F')
 
     const sides = Object.keys(cellStyles.lineWidth)
+    if(sides.length === 4) {
+      const fillColor = cellStyles.fillColor === false ? null : typeof cellStyles.fillColor !== "string" ? 'F' : cellStyles.fillColor;
+      doc.rect(cell.x, cursor.y, cell.width, cell.height, fillColor);
```
Comment (MikeDabrowski) [inline src/tableDrawer.ts:399]: I'm not sure when this line is useful. Maybe need to check the git history to see why it was added.

File: src/tableDrawer.ts [hunk 2]
```diff
@@ -392,9 +392,12 @@ function drawCellBorders(doc: DocHandler, cell: Cell, cursor: Pos) {
       doc.rect(cell.x, cursor.y, cell.width, cell.height, fillStyle)
     }
   } else if (typeof cellStyles.lineWidth === 'object') {
-    doc.rect(cell.x, cursor.y, cell.width, cell.height, 'F')
 
     const sides = Object.keys(cellStyles.lineWidth)
+    if(sides.length === 4) {
+      const fillColor = cellStyles.fillColor === false ? null : typeof cellStyles.fillColor !== "string" ? 'F' : cellStyles.fillColor;
```
Comment (MikeDabrowski) [inline src/tableDrawer.ts:398]: `false` should map to null
non strings to 'F'
strings can be passed to `jsPdf.rect`

File: src/tableDrawer.ts [hunk 3]
```diff
@@ -392,9 +392,12 @@ function drawCellBorders(doc: DocHandler, cell: Cell, cursor: Pos) {
       doc.rect(cell.x, cursor.y, cell.width, cell.height, fillStyle)
     }
   } else if (typeof cellStyles.lineWidth === 'object') {
-    doc.rect(cell.x, cursor.y, cell.width, cell.height, 'F')
 
     const sides = Object.keys(cellStyles.lineWidth)
+    if(sides.length === 4) {
```
Comment (MikeDabrowski) [inline src/tableDrawer.ts:397]: rect basically creates all 4 borders. I added this condition to keep rendering them if less than 4 sides are specified.

File: src/tableDrawer.ts [hunk 4]
```diff
@@ -392,9 +392,12 @@ function drawCellBorders(doc: DocHandler, cell: Cell, cursor: Pos) {
...
+      const fillColor = cellStyles.fillColor === false ? null : typeof cellStyles.fillColor !== "string" ? 'F' : cellStyles.fillColor;
+      doc.rect(cell.x, cursor.y, cell.width, cell.height, fillColor);
+    }
     const lineWidth: any = cellStyles.lineWidth
 
     sides.map((side: string) => {
```
Comment (MikeDabrowski) [inline src/tableDrawer.ts:403]: is there a reason why `Array.map` is used instead `Array.forEach` ? 
Some backward compatibility?

File: src/tableDrawer.ts [hunk 5]
```diff
@@ -382,75 +382,120 @@ function printRow(
...
   doc: DocHandler,
   cell: Cell,
   cursor: Pos,
-  side: string,
-  fillStyle: string,
-  lineWidth: number
+  fillColor: Color
+) {
+  const cellFillColor =
+    fillColor === false ? null : typeof fillColor !== 'string' ? 'F' : fillColor
```
Comment (MikeDabrowski) [inline src/tableDrawer.ts:413]: false should map to null
non strings to 'F'
strings can be passed to jsPdf.rect

**Comment by simonbengtsson** [comment]
Looks great!

**Pull request merged**